### DISCUSSION TITLE
Need help in understanding communication between agent and server

### DISCUSSION THREAD

**Discussion by imtadashi** [Ylianst/MeshCentral#5070] [Q&A] [answered]
Hello, I was going through the application and I am wondering how the communication takes place. I'm interested in understanding the communication process between the server and the agent. Specifically, once the connection is established, I'm curious about how the server is able to communicate with the agent whenever it needs to push JavaScript code or when the server want to access the files or accessing Remote Desktop. Could you please provide me with more detailed information about the underlying mechanism and communication process?

**Comment by silversword411** [comment]
Read the docs: https://ylianst.github.io/MeshCentral/

Watch the videos: https://www.youtube.com/@MeshCentral

**Comment by imtadashi** [comment, marked as answer]
Explanation is given in Issue #5071
